src/darwin/Framework/CHIP/MTRDeviceController.mm

```diff
@@ -793,7 +834,11 @@ - (BOOL)continueCommissioningDevice:(void *)device
         auto deviceProxy = static_cast<chip::DeviceProxy *>(device);
         auto errorCode = self->_cppCommissioner->ContinueCommissioningAfterDeviceAttestation(deviceProxy,
             ignoreAttestationFailure ? chip::Credentials::AttestationVerificationResult::kSuccess : lastAttestationResult);
-        return ![MTRDeviceController checkForError:errorCode logMsg:kErrorPairDevice error:error];
+        auto failed = [MTRDeviceController checkForError:errorCode logMsg:kErrorPairDevice error:error];
```

Comment:
I think the pattern you're using below somewhere of submitting the metric before the `checkForError:` is more readable

Reply:
Which place are you referring to?

Reply:
Found it